### PR TITLE
[ID-62] Feat location permission explaination

### PR DESCRIPTION
**Pull request by DongNguyenQ**

### CODE REVIEW

File: lib/features/home/home_screen.dart [hunk 1]
```diff
@@ -16,103 +19,64 @@ class IDHome extends StatefulWidget {
...
+      body: BlocConsumer<LocationBloc, LocationState>(
+        listener: (_, LocationState state) {
+          if (state is LocationPermissionDeniedState || state is LocationPermissionDeniedForeverState) {
+            context.showIDBottomSheet(
+              iconPath: Assets.images.idLocationImg,
+              title: S.of(context).locationTurnOnSuggestionTitle,
+              content: S.of(context).locationTurnOnSuggestionDescription,
+              primaryButtonText: S.of(context).ctaGoToSettings,
+              textButtonText: S.of(context).ctaLocationTurnOnContinueDefaultLocation,
+              onPrimaryButtonSelected: () => LocationService().requestOpenAppSettings(),
```
Comment (suesitran) [inline lib/features/home/home_screen.dart]: when state is LocationPermissionDeniedState, primary button should send event ShowDialogRequestPermissionEvent()

when state is LocationPermissionDeniedForeverState, primary button should LocationService().requestOpenAppSettings

**Review by suesitran — CHANGES_REQUESTED** [review]
please fix issue in comment

File: lib/features/home/home_screen.dart (hunk 1 above)
Comment (DongNguyenQ) [reply lib/features/home/home_screen.dart]: @suesitran i assumed that you mean 
`state is LocationPermissionDeniedState` -> Fire event ShowDialogRequestPermissionEvent
`state is LocationPermissionDeniedForeverState` -> Show bottom sheet.

is that right ? 

And about requestOpenAppSettings. Do you want me to move it to bloc also for consistency ? 

I have one more question about behaviour:
- If user don't allow permission at first time. We immediately show Bottom Sheet or we only show it the next time user open app ?

Comment (suesitran) [reply lib/features/home/home_screen.dart]: No, in both the states, we show bottom popup, but the behaviours in primary button onPressed are different for each state, and the display text for primary button are different as well. Please check the old code for more details.

Comment (DongNguyenQ) [reply lib/features/home/home_screen.dart]: OLD VERSION 
- `LocationPermissionDeniedState`                        - `LocationPermissionDeniedForeverState`
image                             image




NEW VERSION
- Design : 
image

- The content of new version is different from the old version above. So it gonna be really weird if we keep the content (title, description) for both Denied vs DeniedForever state.
- As i understood. DeniedForever mean that user have denied permission before, and they have to manually  enable it again in device settings. 
- So i changed to this:
- `LocationPermissionDeniedState`
image
- `LocationPermissionDeniedForeverState`
image

Comment (suesitran) [reply lib/features/home/home_screen.dart]: yes, that's correct implementation on the UI now, how about the interaction? 

when `LocationPermissionDeniedState`, and user taps on 'Allow', we need to show native permission request dialog

and when state is `LocationPermissionDeniedForeverState`, when user taps on 'Go to Settings', then we'll open app setting

**Review by suesitran — DISMISSED** [review]
Good job!

**Review by suesitran — APPROVED** [review]

**Pull request merged**